src/tpwm.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * tpwm.c - Uncomprassing TPWM'ed content
+ *
+ * Copyright (C) 2015  Wicked_Digger  <wicked_digger@mail.ru>
+ *
+ * This file is part of freeserf.
+ *
+ * freeserf is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * freeserf is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with freeserf.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "tpwm.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
+#ifdef HAVE_STDINT_H
+# include <stdint.h>
+#endif
+
+static const uint8_t tpwm_sign[4] = {'T','P','W','M'};
+
+#pragma pack(push,1)
+typedef struct {
+	uint8_t magic[4];
+	uint32_t original_size;
+} tpwm_header_t;
+#pragma pack(pop)
```

Comment:
I would not use a packed struct for this. Better add a function that fields directly. You need custom code anyway because you have to convert endianness for the `original_size` field.